### PR TITLE
Aswathy/Account type validation

### PR DESCRIPTION
**Pull request by aswathy-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/signup-affiliates/wizard.tsx [hunk 1]
```diff
@@ -12,15 +14,22 @@ const AffiliateSignup = () => {
         'Personal details',
         'Terms of use',
     ]
+    const [nextaccounttype, setNextAccount] = useState(false)
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/wizard.tsx:17]:
```suggestion
    const [next_account_type, setNextAccount] = useState(false)
```

File: src/pages/signup-affiliates/components/_account-type.tsx [hunk 2]
```diff
@@ -1,45 +1,53 @@
...
 import styled from 'styled-components'
 import IndividualIcon from 'images/svg/signup-affiliates/individual 2.svg'
 import BusinessIcon from 'images/svg/signup-affiliates/company 2.svg'
 import Card, { CardProps } from 'components/custom/_card'
 import { localize, Localize } from 'components/localization'
 import { Header } from 'components/elements'
 
-const AccountType = () => {
+type AccountTypeProps = {
+    setCardSelected: () => void
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/components/_account-type.tsx:10]: What's this function doing?

Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_account-type.tsx:10]: validation

**Review by prince-deriv — APPROVED** [review]

File: src/pages/signup-affiliates/components/_account-type.tsx [hunk 3]
```diff
@@ -1,45 +1,58 @@
...
-            selected: true,
         },
         {
             icon: BusinessIcon,
             title: <Localize translate_text="Business/Corporate" />,
             description: <Localize translate_text="Register as a company or business unit" />,
-            selected: true,
         },
     ]
+    React.useEffect(() => {
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/components/_account-type.tsx:35]: Nitpick can we have a white space here.

File: src/pages/signup-affiliates/components/_account-type.tsx [hunk 4]
```diff
@@ -1,45 +1,58 @@
...
+            {cards.map(({ icon, title, description }, index) => {
                 return (
                     <Card
-                        key={index}
+                        key={`selectedkey-${index}`}
                         icon={icon}
                         title={title}
                         description={description}
-                        selected={selected}
+                        selected={selected_item === index ? true : false}
```
Comment (prince-deriv) [inline src/pages/signup-affiliates/components/_account-type.tsx:51]:
```suggestion
                        selected={selected_item === index}
```

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**